runfile/sleep/sleep-remotehost-runfile.json

```diff
@@ -0,0 +1,93 @@
+{
+    "benchmarks": [
+        {
+            "name": "sleep",
+            "ids": "1",
+            "mv-params": {
+                "global-options": [
+                    {
+                        "name": "global",
+                        "params": [
+                            { "arg": "seconds", "vals": [ "43200" ], "role": "client" }
+                        ]
+                    }
+                ],
+                "sets": [
+                    {
+                        "include": "global",
+                        "params": [
+                        ]
+                    }
+                ]
+            }
+        }
+    ],
+    "tool-params": [
+        {
+            "tool": "procstat",
+            "enabled": "no"
+        },
+        {
+            "tool": "sysstat",
+            "enabled": "no",
+            "params": [
+                { "arg": "subtools", "val": "mpstat" }
+            ]
+        },
+        {
+            "tool": "rt-trace-bpf",
+            "enabled": "no"
+        },
+        {
+            "tool": "kernel",
+            "enabled": "yes",
+            "params": [
+                { "arg": "subtools", "val": "sysfs-trace" },
+                { "arg": "sysfs-trace-setup", "val": "echo ffffffff,ffffffff,ffffffff,fffffffc > tracing_cpumask" },
+                { "arg": "sysfs-trace-setup", "val": "echo 1048576 > buffer_size_kb" },
+                { "arg": "sysfs-trace-setup", "val": "echo function_graph > current_tracer" },
+                { "arg": "sysfs-trace-setup", "val": "echo 0 > max_graph_depth" },
+                { "arg": "sysfs-trace-setup", "val": "echo *:* > set_event" },
+                { "arg": "sysfs-trace-setup", "val": "echo 1 > options/funcgraph-abstime" },
+                { "arg": "sysfs-trace-setup", "val": "echo 1 > options/funcgraph-proc" },
+		{ "arg": "sysfs-trace-setup", "val": "echo 1 > options/latency-format" },
+		{ "arg": "sysfs-trace-setup", "val": "echo 1 > options/funcgraph-tail" },
+                { "arg": "sysfs-trace-cleanup", "val": "echo ffffffff,ffffffff,ffffffff,ffffffff > tracing_cpumask" },
+                { "arg": "sysfs-trace-cleanup", "val": "echo nop > current_tracer" },
+                { "arg": "sysfs-trace-cleanup", "val": "echo 0 > options/funcgraph-abstime" },
+                { "arg": "sysfs-trace-cleanup", "val": "echo 0 > options/funcgraph-proc" },
+		{ "arg": "sysfs-trace-cleanup", "val": "echo 0 > options/latency-format" },
+		{ "arg": "sysfs-trace-cleanup", "val": "echo 0 > options/funcgraph-tail" },
+                { "arg": "sysfs-trace-collection", "val": "per-cpu" }
+            ]
+        }
```

Comment:
Is this actually going to do anything?  My recollection is that this setup stuff is specifically for our modified version of oslat where the oslat process itself will write the `1` to enable the trace and the `0` to disable it.  Since the benchmark can only turn the trace on and off it needs something else to configure the tracer which is what is happening here.  If I'm remembering correctly, having this paired with `bench-sleep` doesn't make much sense...

Reply:
No, this is the tool-kernel: https://github.com/perftool-incubator/tool-kernel/blob/master/kerneltools-start#L43
What you are probably referring to is the oslat exclusive critical sections that can be enabled with:
```
                            { "arg": "trace-control", "vals": [ "1" ], "role": "client", "enabled": "yes" },
                            { "arg": "trace-markers", "vals": [ "1" ], "role": "client", "enabled": "yes" }
```
Running it with sleep would be useful to show the sources of latency on an "idle" system.

Reply:
Right, without this oslat parameter (or something similar from somewhere else):

Reply:
> Right, without this oslat parameter (or something similar from somewhere else):
> 
> ```
> { "arg": "trace-control", "vals": [ "1" ], "role": "client", "enabled": "yes" },
> ```
> 
> there is nothing to actually turn the tracing on and off, so it doesn't do anything.

Wrong. tool-kernel does not have anything to do with the "trace-control" param, which is oslat specific for enabling trace marker from the userspace.

If you just enable the tool-param "kernel", and use these params:
https://github.com/perftool-incubator/tool-kernel/blob/master/kerneltools-start#L31

You can actually enable any tracer available in ftrace.

Does it make sense @k-rister ?